core/silkworm/common/util.cpp

```diff
@@ -166,16 +166,36 @@ std::optional<Bytes> from_hex(std::string_view hex) noexcept {
 
     if (pos) {
         auto b{unhex_lut(static_cast<uint8_t>(*src++))};
-        if (b == 0xbc) {
+        if (b == 0xff) {
             return std::nullopt;
         }
         *dst++ = b;
     }
 
+    // following "while" is unrolling the loop when we have >= 4 target bytes
+    // this is optional, but 5-10% faster
+    while (last - src >= 8) {
+        auto a{unhex_lut4(static_cast<uint8_t>(*src++))};
+        auto b{unhex_lut(static_cast<uint8_t>(*src++))};
+        auto c{unhex_lut4(static_cast<uint8_t>(*src++))};
+        auto d{unhex_lut(static_cast<uint8_t>(*src++))};
+        auto e{unhex_lut4(static_cast<uint8_t>(*src++))};
+        auto f{unhex_lut(static_cast<uint8_t>(*src++))};
+        auto g{unhex_lut4(static_cast<uint8_t>(*src++))};
+        auto h{unhex_lut(static_cast<uint8_t>(*src++))};
+        if ((b | d | f | h) == 0xff || (a | c | e | g) == 0xff ) {
+            return std::nullopt;
+        }
```

Comment:
Can you integrate tests so this condition is covered ?

Reply:
Sure, I just added the test.